src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -136,12 +149,55 @@ public HttpResponse doFinishLogin(StaplerRequest request, StaplerResponse respon
     } catch (RequiresHttpAction e) {
       throw new IllegalStateException(e);
     }
+
     Saml2Profile saml2Profile = client.getUserProfile(credentials, context);
 
-    SamlAuthenticationToken samlAuthToken = new SamlAuthenticationToken(saml2Profile.getId());
+    // retrieve user display name
+    String userFullName = null;
+    List<?> names = (List<?>) saml2Profile.getAttribute(this.displayNameAttributeName);
+    if (names != null && !names.isEmpty()){
+      userFullName = (String)names.get(0);
+    }
+
+    // prepare list of groups
+    List<?> groups = (List<?>) saml2Profile.getAttribute(this.groupsAttributeName);
```

Comment:
It was `List<String>` in the first commit but I changed that because of Java unchecked cast warning.